src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml.cs

```diff
@@ -35,39 +34,16 @@ public ListViewModel? ViewModel
     public static readonly DependencyProperty ViewModelProperty =
         DependencyProperty.Register(nameof(ViewModel), typeof(ListViewModel), typeof(ListPage), new PropertyMetadata(null, OnViewModelChanged));
 
-    public ViewModelLoadedState LoadedState
-    {
-        get => (ViewModelLoadedState)GetValue(LoadedStateProperty);
-        set => SetValue(LoadedStateProperty, value);
-    }
-
-    // Using a DependencyProperty as the backing store for LoadedState.  This enables animation, styling, binding, etc...
-    public static readonly DependencyProperty LoadedStateProperty =
-        DependencyProperty.Register(nameof(LoadedState), typeof(ViewModelLoadedState), typeof(ListPage), new PropertyMetadata(ViewModelLoadedState.Loading));
-
     public ListPage()
     {
         this.InitializeComponent();
     }
 
     protected override void OnNavigatedTo(NavigationEventArgs e)
     {
-        LoadedState = ViewModelLoadedState.Loading;
         if (e.Parameter is ListViewModel lvm)
         {
-            if (!lvm.IsInitialized
-                && lvm.InitializeCommand != null)
-            {
-                ViewModel = null;
-
-                _ = Task.Run(async () => await InitializeViewmodel(lvm));
-            }
-            else
-            {
-                ViewModel = lvm;
-                WeakReferenceMessenger.Default.Send<NavigateToPageMessage>(new(lvm));
```

Comment:
Just for future reference, this is the missing call that's causing us to miss the change of the current page VM when we GoBack.

Reply:
This is now fixed by tracking all navigations of the root frame in the ShellPage.xaml.cs and using that to get the parameter and track the CurrentPage VM centrally there itself.